examples/examples_yaml_test.go

```diff
@@ -74,3 +75,23 @@ func TestSecretsYAML(t *testing.T) {
 		},
 	})
 }
+
+func TestBuildOnPreviewYAML(t *testing.T) {
+	cwd, err := os.Getwd()
+	if !assert.NoError(t, err) {
+		t.FailNow()
+	}
+	var outputBuf bytes.Buffer
+	integration.ProgramTest(t, &integration.ProgramTestOptions{
+		Dir:                      path.Join(cwd, "test-build-on-preview", "yaml"),
+		SkipUpdate:               true, //only run Preview
+		SkipExportImport:         true,
+		Verbose:                  true, //we need this to verify the build output logs
+		AllowEmptyPreviewChanges: true,
```

Comment:
Wouldn't we expect to build an image in the preview?

Reply:
So I think  this flag is poorly named. [All it does is it tells the programtest to _not_ pass the `--expect-no-changes` flag](https://github.com/pulumi/pulumi/blob/master/pkg/testing/integration/program.go#L1509) to `pulumi update`, as the test fails spuriously otherwise. The real check is a few lines below , in whether the preview output contains messages that point to a built image. This test only runs preview, not update.